rpc/src/server/api.rs

```diff
@@ -132,4 +132,21 @@ impl api_server::Api for RpcApi {
 
         self.block_producer.clone().submit_proven_transaction(request).await
     }
+
+    /// Returns details for public (on-chain) account by id.
+    #[instrument(
+        target = "miden-rpc",
+        name = "rpc:get_public_account_details",
+        skip_all,
+        ret(level = "debug"),
+        err
+    )]
+    async fn get_public_account_details(
+        &self,
+        request: tonic::Request<GetPublicAccountDetailsRequest>,
+    ) -> std::result::Result<Response<GetPublicAccountDetailsResponse>, Status> {
+        debug!(target: COMPONENT, request = ?request.get_ref());
+
+        self.store.clone().get_public_account_details(request).await
```

Comment:
I think we should validate the account_id here

Reply:
Done